addons/account/models/account_account.py

```diff
@@ -629,11 +629,18 @@ def _adapt_accounts_for_account_groups(self, account_ids=None):
                    AND agroup.code_prefix_end >= LEFT(account.code, char_length(agroup.code_prefix_end))
                    AND agroup.company_id = account.company_id
                  WHERE account.company_id IN %(company_ids)s {where_account}
+            ),
+            second_relation AS (
+                SELECT account.id as account_id, first_relation.group_id as group_id
+                FROM account_account account
+                LEFT JOIN first_relation
+                ON account.id = first_relation.account_id
+                WHERE account.company_id IN %(company_ids)s {where_account}
             )
             UPDATE account_account account
-               SET group_id = relation.group_id
-              FROM relation
-             WHERE relation.account_id = account.id;
+               SET group_id = second_relation.group_id
+              FROM second_relation
+             WHERE second_relation.account_id = account.id;
```

Comment:
@gawa-odoo Do not you risk putting accounts without group that should have another one? (because I don't think this is recalculating everything)  Also, this is not easy to read.  (it was already not easy before...)

Reply:
wrong tagged person, so I invoke @guva-odoo